src/main/java/com/salesforce/mce/awesolog/S3FixedWindowRollingPolicy.java

```diff
@@ -75,14 +98,15 @@ protected void uploadFileToS3Async(String filename) {
             addInfo("Uploading " + filename);
             Runnable uploader = () -> {
                 try {
-                    PutObjectRequest putObjectRequest = PutObjectRequest
-                        .builder()
-                        .bucket(getS3BucketName())
-                        .key(s3Key)
-                        .build();
                     getS3Client().putObject(
-                        putObjectRequest,
-                        RequestBody.fromFile(file)
```

Comment:
what's wrong with `fromFile`?

Reply:
@realstraw `RequestBody.fromFile(file)` throws `java.io.FileNotFoundException` on the log file to upload to s3 when running as logback extension, and during testing, I checked the paths, whether the `file.exists`, `file.canRead` and the current directory and everything looks good, so far only reading bytes outside and passed in works, I suspect the `FileInputStream` handles the `file` differently to `RequestBody.fromFile`

Reply:
@realstraw just simplified to use `RequestBody.fromInputStream(...)` which also works fine

Reply:
@realstraw ok, the root cause is `fromFile` does a file size check (`Files.size(path)`) which invokes `sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes` and this can throw `java.io.FileNotFoundException` when there's insufficient permission of the user for the `readAttributes`, this explains why it works with either `fromInputStream` or `fromBytes` since these functions don't do file size check, and also this explains why there's no issue running a service with awesolog locally.

Reply:
so `file.length()` works but `File.size(path)` does not?

Reply:
@realstraw correct, `file.length()` goes through `java.io.File` and `File.size(path)` goes through `java.nio.file.Files`